decompiler/frontend/binaryninja/handlers/unary.py

```diff
@@ -0,0 +1,93 @@
+"""Module implementing the UnaryOperationHandler."""
+from functools import partial
+
+from binaryninja import MediumLevelILInstruction, MediumLevelILOperation, mediumlevelil
+from decompiler.frontend.lifter import Handler
+from decompiler.structures.pseudo import BinaryOperation, Constant, Integer, Operation, OperationType, Pointer, UnaryOperation
+
+
+class UnaryOperationHandler(Handler):
+    def register(self):
+        """Register the handling functions at the parent observer."""
+        self._lifter.HANDLERS.update(
+            {
+                mediumlevelil.MediumLevelILNeg: partial(self.lift_unary_operation, OperationType.negate),
+                mediumlevelil.MediumLevelILFneg: partial(self.lift_unary_operation, OperationType.negate),
+                mediumlevelil.MediumLevelILNot: partial(self.lift_unary_operation, OperationType.bitwise_not),
+                mediumlevelil.MediumLevelILSx: self.lift_cast,
+                mediumlevelil.MediumLevelILZx: self._lift_zx_operation,
+                mediumlevelil.MediumLevelILLowPart: self.lift_cast,
+                mediumlevelil.MediumLevelILFloatConv: self.lift_cast,
+                mediumlevelil.MediumLevelILFloatToInt: self.lift_cast,
+                mediumlevelil.MediumLevelILIntToFloat: self.lift_cast,
+                mediumlevelil.MediumLevelILAddressOf: partial(self.lift_unary_operation, OperationType.address),
+                mediumlevelil.MediumLevelILAddressOfField: self.lift_address_of_field,
+                mediumlevelil.MediumLevelILLoad: partial(self.lift_unary_operation, OperationType.dereference),
+                mediumlevelil.MediumLevelILLoadSsa: partial(self.lift_unary_operation, OperationType.dereference),
+                mediumlevelil.MediumLevelILLoadStruct: self._lift_load_struct,
+                mediumlevelil.MediumLevelILLoadStructSsa: self._lift_load_struct,
+                mediumlevelil.MediumLevelILFtrunc: self._lift_ftrunc,
+            }
+        )
+
+    def lift_unary_operation(self, op_type: OperationType, operation: MediumLevelILOperation, **kwargs) -> UnaryOperation:
+        """Lift the given constant value."""
+        return UnaryOperation(
+            op_type,
+            [self._lifter.lift(x, parent=operation) for x in operation.operands],
+            vartype=self._lifter.lift(operation.expr_type, parent=operation),
+        )
+
+    def lift_cast(self, cast: mediumlevelil.MediumLevelILUnaryBase, **kwargs) -> UnaryOperation:
+        """Lift a cast operation, casting one type to another."""
+        return UnaryOperation(OperationType.cast, [self._lifter.lift(cast.src, parent=cast)], vartype=self._lifter.lift(cast.expr_type))
```

Comment:
Why don't we need parent here: `self._lifter.lift(cast.expr_type)` similar to line 38?
Also, question of interest: does type lifting use parent at all?

Reply:
It does not - I think there are only one or two functions using theis value at all
Do you think I should dd the parent value to all lifts of type data for coherence?

Reply:
Puuh... I do not have clear preference here. There are pros and cons of passing `parent` for lifting types, e.g.: pro  - consistency,  con - as you have said, types do not need parent. However, then you need to explain each time someone asks why everyone gets parent during lifting and types do not :D So yeah, it is up to you :)